### DISCUSSION TITLE
Does `command:cmake.buildType` support Multi-Config generators?

### DISCUSSION THREAD

**Discussion by calebkiage** [microsoft/vscode-cmake-tools#2893] [Q&A] [answered]
I have a CMake (presets enabled) project that I'm building on VSCode, I'm using the visual studio generator, but it used some shared DLLs (Catch2). When running the project, I need to add the path to those DLLs for windows to find them. The DLLs are in different locations depending on the build configuration. Setting the environment in the `buildPreset` doesn't seem to be supported by VSCode CMake tools when debugging. I can't use configurePreset since the path changes based on the build type. So, I decided to use custom launch.json configurations. The issue I'm facing is that the `command:cmake.buildType` always resolves to Debug regardless of the actual configuration in the build preset. My question is, does the buildType command support multi-config generators? If not, what is the best approach to take without switching to static builds?

Here is a sample task to test this:
```jsonc
{
    "version": "2.0.0",
    "tasks": [
        {
            "type": "shell",
            "label": "test env",
            "command": "echo",
            "args": [
                "$env:Path"
            ],
            "options": {
                "env": {
                    "PATH": "${env:PATH}C:/dev/libs/${command:cmake.buildType}"
                }
            },
            "problemMatcher": []
        }
    ]
}
```

The path is always `C:/dev/libs/Debug`

**Comment by bobbrow** [comment, marked as answer]
It's a bug in the extension.  It should be addressed with #2895.

**Reply by calebkiage** [reply]
That's great! I have one other related question. Why doesn't the CMake tool use environment variables set in the build presets? It would be useful for this situation so I don't have to use a custom launch.json.
